### PR TITLE
✨ 添加bilibilipush和完善gif文档

### PR DESCRIPTION
**Pull request by guohuiyuan**

### CODE REVIEW

File: plugin/bilibili/bilibilipush.go [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+	data, err := web.RequestDataWith(web.NewDefaultClient(), fmt.Sprintf(infoURL, buid), "GET", referer, ua)
+	if err != nil {
+		return
+	}
+	status = int(gjson.Get(binary.BytesToString(data), "code").Int())
+	name = gjson.Get(binary.BytesToString(data), "data.name").String()
+	if status == 0 {
+		bdb.insertBilibiliUp(buid, name)
+		upMap[buid] = name
+	}
```
Comment (fumiama) [inline plugin/bilibili/bilibilipush.go:188]: else err=...

File: plugin/bilibili/bilibilipush.go [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+
+// subscribe 订阅
+func subscribe(buid, groupid int64) (err error) {
+	bpMap := map[string]interface{}{
+		"bilibili_uid":    buid,
+		"group_id":        groupid,
+		"live_disable":    0,
+		"dynamic_disable": 0,
+	}
+	err = bdb.insertOrUpdateLiveAndDynamic(bpMap)
```
Comment (fumiama) [inline plugin/bilibili/bilibilipush.go:221]: return bdb.insertOrUpdateLiveAndDynamic(bpMap)
其他同理

File: plugin/bilibili/bilibilipushmodel.go [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+	}
+	gdb.AutoMigrate(&bilibilipush{}).AutoMigrate(&bilibiliup{})
+	return (*bilibilipushdb)(gdb)
+}
+
+// insertOrUpdateLiveAndDynamic 插入或更新数据库
+func (bdb *bilibilipushdb) insertOrUpdateLiveAndDynamic(bpMap map[string]interface{}) (err error) {
+	db := (*gorm.DB)(bdb)
+	bp := bilibilipush{}
+	data, _ := json.Marshal(&bpMap)
```
Comment (fumiama) [inline plugin/bilibili/bilibilipushmodel.go:60]: err判断，其他同理

**Review by fumiama — CHANGES_REQUESTED** [review]

**Comment by guohuiyuan** [comment]
已修改



Sent&nbsp;from&nbsp;my&nbsp;iPhone


------------------ 原始邮件 ------------------
发件人: 源文雨 ***@***.***&gt;
发送时间: 2022年8月11日 09:44
收件人: FloatTech/ZeroBot-Plugin ***@***.***&gt;
抄送: himawari ***@***.***&gt;, Author ***@***.***&gt;
主题: Re: [FloatTech/ZeroBot-Plugin] ✨ 添加bilibilipush和完善gif文档 (PR #369)





 
@fumiama requested changes on this pull request.
 
 
In plugin/bilibili/bilibilipush.go:
 &gt; +} +}) +} + +// 储存up的name,uid +func checkBuid(buid int64) (status int, name string, err error) { +data, err := web.RequestDataWith(web.NewDefaultClient(), fmt.Sprintf(infoURL, buid), "GET", referer, ua) +if err != nil { +return +} +status = int(gjson.Get(binary.BytesToString(data), "code").Int()) +name = gjson.Get(binary.BytesToString(data), "data.name").String() +if status == 0 { +bdb.insertBilibiliUp(buid, name) +upMap[buid] = name +}  
else err=...
 
 
In plugin/bilibili/bilibilipush.go:
 &gt; +err = errors.New(msg) +return +} +} +return +} + +// subscribe 订阅 +func subscribe(buid, groupid int64) (err error) { +bpMap := map[string]interface{}{ +"bilibili_uid":    buid, +"group_id":        groupid, +"live_disable":    0, +"dynamic_disable": 0, +} +err = bdb.insertOrUpdateLiveAndDynamic(bpMap)  
return bdb.insertOrUpdateLiveAndDynamic(bpMap)
 其他同理
 
 
In plugin/bilibili/bilibilipushmodel.go:
 &gt; +} +defer f.Close() +} +gdb, err := gorm.Open("sqlite3", dbpath) +if err != nil { +panic(err) +} +gdb.AutoMigrate(&amp;bilibilipush{}).AutoMigrate(&amp;bilibiliup{}) +return (*bilibilipushdb)(gdb) +} + +// insertOrUpdateLiveAndDynamic 插入或更新数据库 +func (bdb *bilibilipushdb) insertOrUpdateLiveAndDynamic(bpMap map[string]interface{}) (err error) { +db := (*gorm.DB)(bdb) +bp := bilibilipush{} +data, _ := json.Marshal(&amp;bpMap)  
err判断，其他同理

**Review by fumiama — APPROVED** [review]

**Pull request merged**